### DISCUSSION TITLE
Devcontainer

### DISCUSSION THREAD

**Discussion by Divelix** [isaac-sim/IsaacLab#298] [Ideas]
Orbit is already VS Code centric, but dev experience leaves much to be desired. Recenly, I started to explore this repo and found for myself that development inside container is most reliable and repeatable option, but current [setup_python_env](https://isaac-orbit.github.io/orbit/source/setup/developer.html) doesn't really work for me (and seems a bit hacky even if it worked).

So my suggestion is to create [devcontainer](https://containers.dev/) ([short intro for VS Code](https://code.visualstudio.com/docs/devcontainers/containers)). It seems like a perfect fit for Orbit. 
Btw, it is possible to use them via docker-compose.yml too (I do so in some projects with multiple devcontainers). 
What do you think about devcontainers for Isaac Orbit?

**Comment by Mayankm96** [comment]
Nice to bring it up again.

We have implemented this feature, but it got sidelined because of other needs. @hhansen-bdai, we should discuss on reviving it up.

**Comment by miguelasd688** [comment]
Maybe it is too late, but.

I recently started working with Isaac Lab and went to find the same answer as @Divelix . I didn't find any so I made a simple example to develop my experiments, this [repo](https://github.com/miguelasd688/isaac-lab-devcontainer) contains all the information to run an standalone IsaacLab installation.

Hope it is useful!
